### PR TITLE
style(proxysql-agent): reorganized the project layout

### PR DESCRIPTION
**Pull request by kuzmik**
- I followed some guides on go project layouts to redo the project layout
- Also enabled some extra golangci checks and addressed as many of the findings I could
- fix: add a configurable namespace to the core pod selector. We have several proxysql namespaces, so being able to configure it at runtime is important
- chore: add goreleaser and workflow

### CODE REVIEW

File: .github/workflows/releaser.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          distribution: goreleaser
+          version: latest
+          args: release --clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (drcapulet) [inline .github/workflows/releaser.yaml:41]: nit: trailing newline in here

File: README.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # ProxySQL Agent
 
 [![Go Report Card](https://goreportcard.com/badge/github.com/kuzmik/proxysql-agent)](https://goreportcard.com/report/github.com/kuzmik/proxysql-agent)
+[![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fkuzmik%2Fproxysql-agent.svg?type=shield)](https://app.fossa.com/projects/git%2Bgithub.com%2Fkuzmik%2Fproxysql-agent?ref=badge_shield)
```
Comment (derektamsen) [inline README.md:4]: I think the fossa path needs to be updated for the persona-id org?

File: build/dev.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+RUN go mod download
+
+COPY . .
+
+RUN apk update \
+    && apk add --no-cache git=2.40.1-r0 \
+    && rm -rf /var/cache/apk/* /lib/apk/db/*
+
+RUN CGO_ENABLED="0" go build -ldflags "-s -w -X 'main.version=${VERSION}' -X 'main.build=${BUILD_SHA}' -X 'main.builddate=${BUILD_TIME}'" -o proxysql-agent cmd/proxysql-agent/main.go
```
Comment (derektamsen) [inline build/dev.Dockerfile:25]: You might not need to specifically inject build information post go 1.18. Looks like a revision and time are automatically added now. https://blog.carlmjohnson.net/post/2023/golang-git-hash-how-to/

Comment (derektamsen) [reply build/dev.Dockerfile:25]: Also, it's probably not good to inject build time as it results in unreproducible builds.

File: .goreleaser.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      {{- else }}{{ .Arch }}{{ end }}
+      {{- if .Arm }}v{{ .Arm }}{{ end }}
+    wrap_in_directory: true
+
+dockers:
+  - dockerfile: build/Dockerfile
+    extra_files:
+      - configs/config.yaml
+    image_templates:
+      - "ghcr.io/kuzmik/{{.ProjectName}}:latest"
```
[the remark this answers is not in the document]
Comment (kuzmik) [reply .goreleaser.yaml:31]: yeah in the next PR in this repo i'll be removing my name in various places and changing it to `persona-id`.

File: README.md (hunk 2 above)
Comment (kuzmik) [reply README.md:4]: that'll be the next pr when i move all the other stuff as well.

**Comment by kuzmik** [comment]
I will replace kuzmik with persona-id as needed in the next PR

**Review by drcapulet — APPROVED** [review]

**Pull request merged**